### PR TITLE
Adapt publisher to dcat ap

### PR DESCRIPTION
**Pull request by sabinem**

### CODE REVIEW

File: ckanext/dcatapchharvest/profiles.py [hunk 1]
```diff
@@ -298,9 +328,10 @@ def parse_dataset(self, dataset_dict, dataset_ref):  # noqa
         )
 
         # Publisher
-        dataset_dict['publishers'] = self._publishers(
+        dataset_dict['publisher'] = self._publisher(
             dataset_ref,
-            DCT.publisher
+            DCT.publisher,
+            dataset_dict['identifier']
```
Comment (kovalch) [inline ckanext/dcatapchharvest/profiles.py:334]: Do we really need to use 'identifier' here, shouldn't it be `organization_url` ?

Comment (kovalch) [reply ckanext/dcatapchharvest/profiles.py:334]: in the migration you use as an url for publisher: ORGANIZATION_URI_BASE + organization_name

Comment (kovalch) [reply ckanext/dcatapchharvest/profiles.py:334]: what happened if an  "identifier" = ""

File: ckanext/dcatapchharvest/tests/fixtures/conformant/dataset-landing-page.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+         xmlns:dcat="http://www.w3.org/ns/dcat#">
+
+  <dcat:Dataset rdf:about="https://swisstopo/123">
+      <dcat:landingPage rdf:resource="https://www.bfs.admin.ch/bfs/de/home/statistiken.html"/>
+  </dcat:Dataset>
+</rdf:RDF>
```
Comment (kovalch) [inline ckanext/dcatapchharvest/tests/fixtures/conformant/dataset-landing-page.xml:8]: it is asked to add a new line at the end of the document

File: ckanext/dcatapchharvest/tests/test_parse_dcatap_conformant_rdf.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        eq_(dataset['url'], u"https://www.bfs.admin.ch/bfs/de/home/statistiken.html")
+
+    def test_dcatap_conformant_publisher_import(self):
+        contents = self._get_file_contents('conformant/dataset-publisher.xml')
+        p = RDFParser(profiles=['swiss_dcat_ap'])
+        p.parse(contents)
+        dataset = [d for d in p.datasets()][0]
+        publisher = json.loads(dataset['publisher'])
+        eq_(publisher['name'], 'Landesamt Topographie Swisstopo')
+        eq_(publisher['url'], 'https://swisstopo')
```
Comment (kovalch) [inline ckanext/dcatapchharvest/tests/test_parse_dcatap_conformant_rdf.py:36]: need to add a new line at the end of the file

File: ckanext/dcatapchharvest/tests/fixtures/deprecated/dataset-publisher.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+         xmlns:dcat="http://www.w3.org/ns/dcat#"
+         xmlns:dct="http://purl.org/dc/terms/"
+         xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
+
+  <dcat:Dataset rdf:about="https://swisstopo/123">
+    <dct:publisher>
+      <rdf:Description>
+        <rdf:type rdf:resource="http://www.w3.org/1999/02/22-rdf-syntax-ns#Description"/>
+        <rdfs:label>Bundesamt für Landestopografie swisstopo</rdfs:label>
+      </rdf:Description>
```
Comment (kovalch) [inline ckanext/dcatapchharvest/tests/fixtures/deprecated/dataset-publisher.xml:12]: you check here for the old format of publishers and use `lable` to create a new format, right?
as is given in ` publisher_deprecated = self._object_value(agent, RDFS.label)`

If so, why do we already have ` <dct:publisher>` here and not  `<dct:publishers>`?

File: ckanext/dcatapchharvest/tests/test_parse_deprecated_rdf.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        eq_(dataset['url'], u"https://www.bfs.admin.ch/bfs/de/home/statistiken.html")
+
+    def test_deprecated_publisher_import(self):
+        contents = self._get_file_contents('deprecated/dataset-landing-page.xml')
+        p = RDFParser(profiles=['swiss_dcat_ap'])
+        p.parse(contents)
+        dataset = [d for d in p.datasets()][0]
+        publisher = json.loads(dataset['publisher'])
+        eq_(publisher['name'], 'Landesamt Topographie Swisstopo')
+        eq_(publisher['url'], 'https://swisstopo')
```
Comment (kovalch) [inline ckanext/dcatapchharvest/tests/test_parse_deprecated_rdf.py:36]: It is asking to add a new line at the end of the document

File: ckanext/dcatapchharvest/tests/test_parse_deprecated_rdf.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+class DeprecatedProfileParseTest(BaseParseTest):
+    def test_deprecated_landing_page_import(self):
+        contents = self._get_file_contents('deprecated/dataset-landing-page.xml')
+        p = RDFParser(profiles=['swiss_dcat_ap'])
+        p.parse(contents)
+        dataset = [d for d in p.datasets()][0]
+        eq_(dataset['url'], u"https://www.bfs.admin.ch/bfs/de/home/statistiken.html")
+
+    def test_deprecated_publisher_import(self):
+        contents = self._get_file_contents('deprecated/dataset-landing-page.xml')
```
Comment (kovalch) [inline ckanext/dcatapchharvest/tests/test_parse_deprecated_rdf.py:30]: don't you want to use `deprecated/dataset-publisher.xml ` here?

File: ckanext/dcatapchharvest/profiles.py (hunk 1 above)
Comment (sabinem) [reply ckanext/dcatapchharvest/profiles.py:334]: Yes that is true, but I noticed that at the time we parse the datasets we don't have the organisation content. It is always just derived from the identifier and added later in the harvesting process not during the parsing, see here: https://github.com/opendata-swiss/ckanext-dcatapchharvest/blob/cee6654127ca5b28dc301783ccad71e8b12781e7/ckanext/dcatapchharvest/harvesters.py#L83

**Review by kovalch — APPROVED** [review]

**Pull request merged**